### PR TITLE
add tiflash label settings example

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add tiflash label settings example

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -223,6 +223,9 @@ delta_index_cache_size = 0
 
 除以上几项外，其余功能参数和 TiKV 的配置相同。需要注意的是：`key` 为 `engine` 的 `label` 是保留项，不可手动配置。
 
+### 通过拓扑 label 进行副本调度
+[TiFlash 设置可用区](https://docs.pingcap.com/zh/tidb/stable/use-tiflash#%E8%AE%BE%E7%BD%AE%E5%8F%AF%E7%94%A8%E5%8C%BA)
```
Comment (Lloyd-Pottiger) [inline tiflash/tiflash-configuration.md:227]:
```suggestion

[TiFlash 设置可用区](https://docs.pingcap.com/zh/tidb/stable/use-tiflash#%E8%AE%BE%E7%BD%AE%E5%8F%AF%E7%94%A8%E5%8C%BA)
```

File: schedule-replicas-by-topology-labels.md [hunk 2]
```diff
@@ -38,6 +38,20 @@ tikv-server --labels zone=<zone>,rack=<rack>,host=<host>
 labels = "zone=<zone>,rack=<rack>,host=<host>"
 ```
 
+TiFlash 同样支持。
+
+{{< copyable "" >}}
+
+```toml
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:45]: 在 152 行的 tiup 例子里面也加上 tiflash label 的配置示例

File: schedule-replicas-by-topology-labels.md [hunk 3]
```diff
@@ -38,6 +38,20 @@ tikv-server --labels zone=<zone>,rack=<rack>,host=<host>
 labels = "zone=<zone>,rack=<rack>,host=<host>"
 ```
 
+TiFlash 同样支持。
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:41]:
```suggestion
TiFlash 支持使用配置文件的方式：
```

Comment (JaySon-Huang) [reply schedule-replicas-by-topology-labels.md:41]: 18 行改为：设置 TiKV/TiFlash 的 `labels` 配置

File: config-templates/complex-tiflash.yaml [hunk 4]
```diff
@@ -124,6 +124,10 @@ tiflash_servers:
     #   # storage.latest.capacity: [ 161061273600 ]
     # learner_config:
     #   log-level: "info"
+    #   server.labels:
+    #     zone: "z2"
+    #     dc: "dc2"
+    #     host: "host2"
```
Comment (JaySon-Huang) [inline config-templates/complex-tiflash.yaml:130]: Maybe keep the same format as tikv does in line 95?
```suggestion
    #   server.labels: { zone: "zone1", dc: "dc1", host: "host1" }
```

File: schedule-replicas-by-topology-labels.md [hunk 5]
```diff
@@ -150,6 +164,51 @@ tikv_servers:
...
+  - host: tiflash-7
+    learner_config:
+      server.labels:
+        zone: z6
+        host: h1
+  - host: tiflash-8
+    learner_config:
+      server.labels:
+        zone: z6
+        host: h2
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:211]:
```suggestion
# z1
  - host: tiflash-1
    learner_config:
      server.labels:
        zone: z1
        host: h1
   - host: tiflash-2
    learner_config:
      server.labels:
        zone: z1
        host: h1
  - host: tiflash-3
    learner_config:
      server.labels:
        zone: z1
        host: h2
  - host: tiflash-4
    learner_config:
      server.labels:
        zone: z1
        host: h2
# z2
  - host: tiflash-5
    learner_config:
      server.labels:
        zone: z2
        host: h1
   - host: tiflash-6
    learner_config:
      server.labels:
        zone: z2
        host: h2
# z3
  - host: tiflash-7
    learner_config:
      server.labels:
        zone: z3
        host: h1
  - host: tiflash-8
    learner_config:
      server.labels:
        zone: z3
        host: h2
```

Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md:211]: deploy tiflash and  tikv in different zone.

File: schedule-replicas-by-topology-labels.md [hunk 6]
```diff
@@ -97,9 +111,9 @@ pd-ctl config set isolation-level zone
 
 ### 使用 TiUP 进行配置（推荐）
 
-如果使用 TiUP 部署集群，可以在[初始化配置文件](/production-deployment-using-tiup.md#第-3-步初始化集群拓扑文件)中统一进行 location 相关配置。TiUP 会负责在部署时生成对应的 TiKV 和 PD 配置文件。
+如果使用 TiUP 部署集群，可以在[初始化配置文件](/production-deployment-using-tiup.md#第-3-步初始化集群拓扑文件)中统一进行 location 相关配置。TiUP 会负责在部署时生成对应的 TiKV、PD 和 TiFlash 配置文件。
 
-下面的例子定义了 `zone/host` 两层拓扑结构。集群的 TiKV 分布在三个 zone，每个 zone 内有两台主机，其中 z1 每台主机部署两个 TiKV 实例，z2 和 z3 每台主机部署 1 个实例。以下例子中 `tikv-n` 代表第 n 个 TiKV 节点的 IP 地址。
+下面的例子定义了 `zone/host` 两层拓扑结构。集群的 TiKV 分布在三个 zone，每个 zone 内有两台主机，其中 z1 每台主机部署两个 TiKV 实例，z2 和 z3 每台主机部署 1 个实例。以下例子中 `tikv-n` 代表第 n 个 TiKV 节点的 IP 地址。集群的 TiFlash 分布在三个 zone，每个 zone 内有两台主机，其中 z4 每台主机部署两个 TiFlash 实例，z5 和 z6 每台主机部署 1 个实例。以下例子中 `tiflash-n` 代表第 n 个 TiFlash 节点的 IP 地址。
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:116]:
```suggestion
下面的例子定义了 `zone/host` 两层拓扑结构。集群的 TiKV 分布在三个 zone，每个 zone 内有两台主机，其中 z1 每台主机部署两个 TiKV 实例，z2 和 z3 每台主机部署 1 个实例。以下例子中 `tikv-n` 代表第 n 个 TiKV 节点的 IP 地址。集群的 TiFlash 分布在三个 zone，每个 zone 内有两台主机，每台主机部署 1 个实例。以下例子中 `tiflash-n` 代表第 n 个 TiFlash 节点的 IP 地址。
```
tiflash 一般不建议单机器多实例部署

File: config-templates/complex-tiflash.yaml (hunk 4 above)
Comment (Lloyd-Pottiger) [reply config-templates/complex-tiflash.yaml:130]: > Ansible doesn’t really use these too much.
> https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html

Multiple lines representations are more recommended in yaml.

Comment (JaySon-Huang) [reply config-templates/complex-tiflash.yaml:130]: Maybe align the style of tikv to tiflash? Or it maybe confusing that two style in the same yaml file.

File: schedule-replicas-by-topology-labels.md (hunk 5 above)
Comment (JaySon-Huang) [reply schedule-replicas-by-topology-labels.md:211]: 为什么需要 tiflash 和 tikv 在不同的 zone？感觉实际中一般比较少这样的需求

Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md:211]: 那我改成每个zone四台机器？tiflash 和 tikv 部署在一台机器也不合适？

Comment (JaySon-Huang) [reply schedule-replicas-by-topology-labels.md:211]: 嗯，每个 zone 四台机器，两台部署 tikv，两台部署 tiflash 这样吧。

File: config-templates/complex-tiflash.yaml [hunk 7]
```diff
@@ -124,6 +127,10 @@ tiflash_servers:
     #   # storage.latest.capacity: [ 161061273600 ]
     # learner_config:
     #   log-level: "info"
+    #   server.labels:
+    #     zone: "z2"
```
Comment (JaySon-Huang) [inline config-templates/complex-tiflash.yaml:131]:
```suggestion
    #     zone: "zone2"
```
aligned to line 96

File: schedule-replicas-by-topology-labels.md [hunk 8]
```diff
@@ -150,6 +164,41 @@ tikv_servers:
...
+  - host: tiflash-7
+    learner_config:
+      server.labels:
+        zone: z3
+        host: h3
+  - host: tiflash-8
+    learner_config:
+      server.labels:
+        zone: z3
+        host: h4
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:201]:
```suggestion
  - host: tiflash-3
    learner_config:
      server.labels:
        zone: z2
        host: h3
   - host: tiflash-4
    learner_config:
      server.labels:
        zone: z2
        host: h4
# z3
  - host: tiflash-5
    learner_config:
      server.labels:
        zone: z3
        host: h3
  - host: tiflash-6
    learner_config:
      server.labels:
        zone: z3
        host: h4
```

File: schedule-replicas-by-topology-labels.md [hunk 9]
```diff
@@ -38,6 +38,20 @@ tikv-server --labels zone=<zone>,rack=<rack>,host=<host>
...
+
+```toml
+[learner_config]
+[learner_config.server]
+[learner_config.server.labels]
+zone="<zone>"
+dc="<dc>"
+rack="<rack>"
+host="<host>"
+```
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:53]:
```suggestion
TiFlash 支持使用 tiflash_tikv.toml 配置文件的方式：

{{< copyable "" >}}

```toml
[server]
labels = "zone=<zone>,rack=<rack>,host=<host>"
```
```
不通过 tiup，手动配置的话，应该是这样？

Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md:53]: toml 格式是这样的
```toml
[learner_config]
[learner_config.server]
```
这两行可以不写，但我看其他配置是这样写的，为了统一风格就加上了。

Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md:53]: image

Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md:53]: image

Comment (Lloyd-Pottiger) [inline schedule-replicas-by-topology-labels.md:53]:
```suggestion
TiFlash 支持使用 tiflash-learner.toml 配置文件的方式：

{{< copyable "" >}}

```toml
[server]
[server.labels]
zone="<zone>"
dc="<dc>"
rack="<rack>"
host="<host>"
```
```

File: schedule-replicas-by-topology-labels.md [hunk 10]
```diff
@@ -35,7 +35,24 @@ tikv-server --labels zone=<zone>,rack=<rack>,host=<host>
...
 [server]
-labels = "zone=<zone>,rack=<rack>,host=<host>"
+[server.labels]
+zone = "<zone>"
+dc = "<dc>"
+rack = "<rack>"
+host = "<host>"
+```
+
+TiFlash 支持使用配置文件的方式：
```
Comment (JaySon-Huang) [inline schedule-replicas-by-topology-labels.md:45]:
```suggestion
TiFlash 支持通过 tiflash-proxy 配置文件的方式设置 labels：
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Comment by JaySon-Huang** [comment]
是不是应该也 pick 到 5.x 分支？

**Review by flowbehappy — APPROVED** [review]

File: tiup/tiup-cluster.md [hunk 11]
```diff
@@ -193,19 +201,22 @@ tiup cluster display prod-cluster
...
+172.16.5.140:4000   tidb        172.16.5.140  4000/10080                       linux/x86_64  Up      -                     deploy/tidb-4000
+172.16.5.141:9000   tiflash     172.16.5.141  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.142:9000   tiflash     172.16.5.142  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.143:9000   tiflash     172.16.5.143  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.134:20160  tikv        172.16.5.134  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
+172.16.5.139:20160  tikv        172.16.5.139  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
+172.16.5.140:20160  tikv        172.16.5.140  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
 ```
 
 Status 列用 `Up` 或者 `Down` 表示该服务是否正常。对于 PD 组件，同时可能会带有 `|L` 表示该 PD 是 Leader，`|UI` 表示该 PD 运行着 [TiDB Dashboard](/dashboard/dashboard-intro.md)。
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tiup/tiup-cluster.md:222]: I didn't found either 😄 , how about open an new issue and ask the pd team to add description about it in `dashboard/dashboard-cluster-info.md`?

[the remark this answers is not in the document]
Comment (shichun-0415) [reply tiup/tiup-cluster.md:222]: > I didn't found either 😄 , how about open an new issue and ask the pd team to add description about it in `dashboard/dashboard-cluster-info.md`?

LGTM

File: tiflash/use-tiflash.md [hunk 12]
```diff
@@ -152,15 +152,26 @@ SELECT TABLE_NAME FROM information_schema.tables where TABLE_SCHEMA = "<db_name>
...
       - host: 172.16.5.85
         config:
-          flash.proxy.labels: zone=z2
+          logger.level: "info"
+        learner_config:
+          server.labels:
+            zone: "z2"
     ```
 
+    注：旧版本中的 `flash.proxy.labels` 配置无法处理可用区名字中的特殊字符，建议使用 `learner_config` 中的 `server.labels` 来进行配置。
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tiflash/use-tiflash.md:173]: The version before modification. We want to keep this in order to remain user that the previous docs is not accurate.

File: schedule-replicas-by-topology-labels.md [hunk 13]
```diff
@@ -15,7 +15,7 @@ aliases: ['/docs-cn/dev/schedule-replicas-by-topology-labels/','/docs-cn/dev/how
 
 ## 根据集群拓扑配置 labels
 
-### 设置 TiKV 的 `labels` 配置
+### 设置 TiKV/TiFlash 的 `labels` 配置
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply schedule-replicas-by-topology-labels.md]: 1,2 have been solved. For 3, tiflash do not support this way.

File: tiflash/use-tiflash.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tiflash/use-tiflash.md:173]: cc @flowbehappy

File: tiup/tiup-cluster.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply tiup/tiup-cluster.md:222]: #10398 done. Please contact pd team to fix it.

Comment (Lloyd-Pottiger) [inline tiup/tiup-cluster.md:222]:
```suggestion
Status 列用 `Up` 或者 `Down` 表示该服务是否正常。对于 PD 组件，同时可能会带有 `|L` 表示该 PD 是 Leader，`|UI` 表示该 PD 运行着 [TiDB Dashboard](dashboard/dashboard-cluster-info.md)。
```

File: tiup/tiup-cluster.md [hunk 14]
```diff
@@ -193,22 +201,25 @@ tiup cluster display prod-cluster
...
+172.16.5.140:4000   tidb        172.16.5.140  4000/10080                       linux/x86_64  Up      -                     deploy/tidb-4000
+172.16.5.141:9000   tiflash     172.16.5.141  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.142:9000   tiflash     172.16.5.142  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.143:9000   tiflash     172.16.5.143  9000/8123/3930/20170/20292/8234  linux/x86_64  Up      data/tiflash-9000     deploy/tiflash-9000
+172.16.5.134:20160  tikv        172.16.5.134  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
+172.16.5.139:20160  tikv        172.16.5.139  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
+172.16.5.140:20160  tikv        172.16.5.140  20160/20180                      linux/x86_64  Up      data/tikv-20160       deploy/tikv-20160
+```
+
+Status 列用 `Up` 或者 `Down` 表示该服务是否正常。对于 PD 组件，同时可能会带有 `|L` 表示该 PD 是 Leader，`|UI` 表示该 PD 运行着 [TiDB Dashboard](dashboard/dashboard-cluster-info.md)。
```
Comment (shichun-0415) [inline tiup/tiup-cluster.md:222]:
```suggestion
Status 列用 `Up` 或者 `Down` 表示该服务是否正常。对于 PD 组件，同时可能会带有 `|L` 表示该 PD 是 Leader，`|UI` 表示该 PD 运行着 [TiDB Dashboard](/dashboard/dashboard-intro.md)。
```

**Comment by Lloyd-Pottiger** [comment]
@shichun-0415 thank you~ Could you please help merge this pr?

**Comment by shichun-0415** [comment]
@Lloyd-Pottiger The original use tiflash.md has been split into several docs in https://github.com/pingcap/docs-cn/pull/10424. Therefore, I synchronized the change in use tiflash.md to create tiflash replicas.md. Please take a look at this PR to ensure that every change is correct. Thanks.

**Comment by Lloyd-Pottiger** [comment]
> @Lloyd-Pottiger The original use tiflash.md has been split into several docs in #10424. Therefore, I synchronized the change in use tiflash.md to create tiflash replicas.md. Please take a look at this PR to ensure that every change is correct. Thanks.

LGTM

File: schedule-replicas-by-topology-labels.md [hunk 15]
```diff
@@ -15,18 +15,18 @@ aliases: ['/docs-cn/dev/schedule-replicas-by-topology-labels/','/docs-cn/dev/how
 
 ## 根据集群拓扑配置 labels
 
-### 设置 TiKV 的 `labels` 配置
+### 设置 TiKV 和 TiFlash 的 `labels`
 
-TiKV 支持在命令行参数或者配置文件中以键值对的形式绑定一些属性，我们把这些属性叫做标签 (label)。TiKV 在启动后，会将自身的标签上报给 PD，因此我们可以使用标签来标识 TiKV 节点的地理位置。
+TiKV 和 TiFlash 支持在命令行参数或者配置文件中以键值对的形式绑定一些属性，我们把这些属性叫做标签 (label)。TiKV 和 TiFlash 在启动后，会将自身的标签上报给 PD，因此我们可以使用标签来标识 TiKV 和 TiFlash 节点的地理位置。
```
Comment (shichun-0415) [inline schedule-replicas-by-topology-labels.md:20]:
```suggestion
TiKV 和 TiFlash 支持在命令行参数或者配置文件中以键值对的形式绑定一些属性，我们把这些属性叫做标签 (label)。TiKV 和 TiFlash 在启动后，会将自身的标签上报给 PD，因此可以使用标签来标识 TiKV 和 TiFlash 节点的地理位置。
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**